talk to servers on non-standard https port (like -s acme.example.org:8443)

ThomasWaldmann opened the pull request:
note:

I removed Network.server attribute, it was not used (only .server_url is used).

I also removed the sanity-check for the ACME server (so one can use server:port syntax, which was considered invalid).
I don't think this sanity-check is needed, because in the end, we can't fully check if it is correct anyway (you can always
give valid, but non-working server names or ports).
Also, many people will use the default ACME server, which is builtin and correct.

tested & works.

jdkasten commented:
This looks good @ThomasWaldmann.  After the quick documentation switch... I think it is good to go.

I should make a slight change in how I store recovery tokens, but it shouldn't affect this PR.

ThomasWaldmann commented:
OK, PR updated.

Note: i kept the ivar docs at class level (== same as at other places in the code) and just removed the .server ivar docstring and added a server param docstring to __init__.

kuba commented on letsencrypt/scripts/main.py:
```diff
@@ -30,7 +30,7 @@ def main():
                         nargs="+")
     parser.add_argument("-s", "--server", dest="server",
-                        help="The ACME CA server address.")
+                        help="The ACME CA server [%s]." % CONFIG.ACME_SERVER)
```
please use `default=CONFIG.ACME_SERVER` instead

ThomasWaldmann replied:
ok, done.

jdkasten commented:
Thanks @ThomasWaldmann!
